deployments/cardano-nodes-env-staging.nix

```diff
@@ -7,12 +7,12 @@ let nodeMap = globals.nodeMap; in
 {
   network.description = "Cardano Staging";
 
-  resources = {
+  resources = { config }: {
     elasticIPs = nodesElasticIPs nodeMap;
     datadogMonitors = (with (import ./../modules/datadog-monitors.nix); {
       cpu = mkMonitor (cpu_monitor // {
         message = pagerDutyPolicy.nonCritical;
-        query = config: "avg(last_5m):avg:system.load.norm.1{env:${config.deployment.name}} by {host} > 0.99";
+        query = config: "avg(last_5m):avg:system.load.norm.1{env:${environment},depl:${config.deployment.name}} by {host} > 0.99";
```

Comment:
Is the intent to deploy these changes to staging for verification and then, if it works, go back and apply the equivalent changes to `deployments/cardano-nodes-env-production.nix`?

Reply:
@jmitchell, the `-production` has no comparable code -- this was what I inherited.

Reply:
Do we need `env:${environment}` if we already filter by deployment name?

Reply:
@domenkozar, we want flexibility in subsetting.  Also, I think in the past, one of you guys argued for keeping `env:${environment}`.

Reply:
I'm all for having both tags, but I don't think it makes sense to filter this query on both. You should either have a monitor per deployment (that would be nice, to be able to mute it) or per env, but not filter for both.

Reply:
@domenkozar, some monitors should be used across all "production" environments, whether they're mainnet, testnet, infra, etc. A good example is the disk utilization monitor. We could just duplicate those monitors across all those envs, but it seems less error prone to use two tag types: one for deployment context (e.g. mainnet, testnet, staging, infra) and one to indicate severity (e.g. production/customer-impacting and internal). This would also enable monitors to distinguish the mainnet deployer from the staging deployer, both of which are infra, and set the escalation level accordingly.

Reply:
We are still talking about two different things. What you and Serge refer to is the following line:

`services.dd-agent.tags = ["env:production", "depl:${config.deployment.name}"];`

That's good, actually we talked about setting this weeks ago.

The part that I'm confused is the above 15 line where we configure the monitor to filter on BOTH tags. My question is why both for this monitor that belong only to staging cardano nodes? (also `environment` shouldn't be used as a variable, but `staging` should be hardcoded as it makes no sense to have a var.

Reply:
@domenkozar, I think we want @jmitchell to tell us how he wants the monitors to be set up.. I don't have an argument either way.

Reply:
Understood, I don't think this will break anythinig, it's just about being explicit what we want. Willing to merge if you can address the rest of the changes